src/uct/ib/base/ib_md.c

```diff
@@ -1439,28 +1464,28 @@ static double uct_ib_md_read_pci_bw(struct ibv_device *ib_device)
         return DBL_MAX;
     }
 
-    if ((sscanf(pci_speed_str, "%lf%s", &bw, gts) < 2) ||
+    if ((sscanf(pci_speed_str, "%lf%s", &bw_gbps, gts) < 2) ||
         strcasecmp("GT/s", ucs_strtrim(gts))) {
         ucs_debug("incorrect format of %s file: expected: <double> GT/s, actual: %s\n",
                   pci_speed_file_name, pci_speed_str);
         return DBL_MAX;
     }
 
-    bw *= UCS_GBYTE / 8; /* gigabit -> gigabyte */
-
     for (i = 0; i < ucs_static_array_size(uct_ib_md_pci_info); i++) {
-        if (bw < (uct_ib_md_pci_info[i].bw * 1.2)) { /* use 1.2 multiplex to avoid round issues */
-            p = &uct_ib_md_pci_info[i]; /* use pointer to make equation shorter */
-            /* coverity[overflow] */
-            effective_bw = bw * width *
-                           (p->payload * p->nack) /
-                           (((p->payload + p->overhead) * p->nack) + p->ctrl) *
-                           p->encoding / p->decoding;
-            ucs_trace("%s: pcie %ux %s, effective throughput %.3lfMB/s (%.3lfGb/s)",
-                      ib_device->name, width, p->name,
-                      (effective_bw / UCS_MBYTE), (effective_bw * 8 / UCS_GBYTE));
-            return effective_bw;
+        p = &uct_ib_md_pci_info[i];
+        if ((int)bw_gbps != p->bw_gbps) {
+            continue;
         }
+
+        link_utilization = (double)(p->payload * p->ctrl_ratio) /
+                           (((p->payload + p->tlp_overhead) * p->ctrl_ratio) +
+                            p->ctrl_overhead);
+        effective_bw     = (p->bw_gbps * 1e9 / 8.0) * width *
+                           ((double)p->encoding / p->decoding) * link_utilization;
+        ucs_trace("%s: PCIe %s %ux, effective throughput %.3f MB/s %.3f Gb/s",
+                  ib_device->name, p->name, width, effective_bw / UCS_MBYTE,
```

Comment:
pls, use either `UCS_MBYTE`/`UCS_GBYTE` or `1e-6`/`1e-9`
and above for `1e9`

Reply:
it's intentional: when showing MBytes/sec use powers of 1024, byt when should Gbits it's powers of 1000

Reply:
@yosefe I see, thanks!